<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CameraView</class>
 <widget class="QWidget" name="CameraView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>858</width>
    <height>553</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>相机</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_6">
   <item row="1" column="1">
    <widget class="QFrame" name="frame">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QPushButton" name="pushButton_CameraOn">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>摄像机开</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pushButton_CameraOff">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>摄像机关</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_Focus">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Focus</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_Shot">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>拍照</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>曝光补偿</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QSlider" name="horizontalSlider_exposureCompensation">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>170</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="minimum">
         <number>-4</number>
        </property>
        <property name="maximum">
         <number>4</number>
        </property>
        <property name="pageStep">
         <number>2</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>图片质量</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QSlider" name="horizontalSlider_Quality">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>170</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::NoContextMenu</enum>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QGroupBox" name="groupBox">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>摄像头</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QComboBox" name="comboBox_CameraDevice">
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>170</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QGroupBox" name="groupBox_4">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>分辨率</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QComboBox" name="comboBox_Resolution">
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>170</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QGroupBox" name="groupBox_5">
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="title">
      <string>图片格式</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QComboBox" name="comboBox_Codec">
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>170</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="7" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0" rowspan="7">
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QCameraViewfinder" name="viewfinder" native="true">
        <property name="minimumSize">
         <size>
          <width>640</width>
          <height>480</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QFrame" name="frame_4">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QPushButton" name="pushButton_FindSavePath">
        <property name="text">
         <string>保存到</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_SavePath">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>计数器:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdNumber"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCameraViewfinder</class>
   <extends>QWidget</extends>
   <header>qcameraviewfinder.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_FindSavePath</sender>
   <signal>clicked()</signal>
   <receiver>CameraView</receiver>
   <slot>OnFindSavePath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_CameraOn</sender>
   <signal>clicked()</signal>
   <receiver>CameraView</receiver>
   <slot>OnCameraPowerOn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>569</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_CameraOff</sender>
   <signal>clicked()</signal>
   <receiver>CameraView</receiver>
   <slot>OnCameraPowerOff()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>665</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>691</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Focus</sender>
   <signal>clicked()</signal>
   <receiver>CameraView</receiver>
   <slot>OnFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>552</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Shot</sender>
   <signal>clicked()</signal>
   <receiver>CameraView</receiver>
   <slot>OnShot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>712</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_exposureCompensation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CameraView</receiver>
   <slot>OnValueChanged_ExposureCompensation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>602</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_Quality</sender>
   <signal>valueChanged(int)</signal>
   <receiver>CameraView</receiver>
   <slot>OnValueChanged_Quality(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>505</x>
     <y>242</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_CameraDevice</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>CameraView</receiver>
   <slot>OnSelectedChanged_CameraDevice(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>548</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Resolution</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>CameraView</receiver>
   <slot>OnSelectedChanged_Resolution(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>565</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Codec</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>CameraView</receiver>
   <slot>OnSelectedChanged_Codec(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>547</x>
     <y>412</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>418</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnFindSavePath()</slot>
  <slot>OnCameraPowerOn()</slot>
  <slot>OnCameraPowerOff()</slot>
  <slot>OnFocus()</slot>
  <slot>OnShot()</slot>
  <slot>OnValueChanged_ExposureCompensation(int)</slot>
  <slot>OnValueChanged_Quality(int)</slot>
  <slot>OnSelectedChanged_CameraDevice(QString)</slot>
  <slot>OnSelectedChanged_Resolution(QString)</slot>
  <slot>OnSelectedChanged_Codec(QString)</slot>
 </slots>
</ui>
